src/main/java/org/springframework/data/jpa/repository/query/JpaQueryExecution.java

```diff
@@ -298,4 +309,48 @@ protected Object doExecute(AbstractJpaQuery jpaQuery, Object[] values) {
 			return storedProcedureJpaQuery.extractOutputValue(storedProcedure);
 		}
 	}
+
+	static class StreamExecution extends JpaQueryExecution {
+
+		@Override
+		protected Object doExecute(AbstractJpaQuery query, Object[] values) {
+
+			Query jpaQuery = query.createQuery(values);
```

Comment:
This is just a raw sketch .. for a real implementation we need to abstract the Provider details ...